src/youtube_chapters_demo.py

```diff
@@ -0,0 +1,177 @@
+    else:
+        date_time = datetime.datetime.strptime(time_stamp, "%M:%S")
+    
+    a_timedelta = date_time - datetime.datetime(1900, 1, 1)
+    seconds = a_timedelta.total_seconds()
+    return seconds 
+
+
+def create_markers(chapters, fps):
+    markers = []
+
+    for chapter in chapters: 
+        seconds = convert_time_stamp_to_seconds(chapter[0])
+ 
+        marker = otio.schema.Marker()
+
+        marker.marked_range = otio.opentime.TimeRange(
+            start_time=otio.opentime.RationalTime(seconds*fps, fps),
+            duration=otio.opentime.RationalTime() #TODO: Set a specific duration
+        )
+ 
+        marker.color = otio.schema.MarkerColor.RED
+ 
+        marker.name = chapter[1]
+        markers.append(marker)
+
+
+    return markers
+
+
+
+
+
+def download_from_youtube(youtubeURL): 
+    ydl_opts = {
+        'outtmpl': 'tmp/%(id)s.mp4',
+        'noplaylist': True,
+        'quiet': True,
+        'writedescription' : True
+    }
+
+    # Download the youtube video and description onto local computer 
+    with youtube_dl.YoutubeDL(ydl_opts) as ydl:
+        dictMeta = ydl.extract_info(
+            "https://www.youtube.com/watch?v={sID}".format(sID=youtubeURL),
+            download=True)
+    
+    return dictMeta
+    
+
+
+def create_timeline(dictMeta, video_file_name): 
+    # Steps for importing the video into OTIO
+    # 1. Create a TimeLine object
+    timeline = otio.schema.Timeline()
+    timeline.name = "Youtube Demo"
+
+    # 2. Create a Track on the timeline 
+    track = otio.schema.Track()
+    track.name = "Videos"
+    timeline.tracks.append(track)
+
+    # 3. Find out how long the youtube video is
+    totalFrames = dictMeta['duration'] * dictMeta['fps']
+
+
+    available_range = otio.opentime.TimeRange(
+        otio.opentime.RationalTime(0, dictMeta['fps']),
+        otio.opentime.RationalTime(totalFrames, dictMeta['fps'])
+    )
+
+
+    # 4. Create a media_reference (contians the file path of the youtube video)
+
+    media_reference = otio.schema.ExternalReference(
+            target_url="tmp/" + video_file_name,
+            available_range=available_range
+        )
+
+    # 5. Create a Clip and set the media_reference (based on what we found in step 4) 
+    clip = otio.schema.Clip(name="Youtube clip")
+    clip.media_reference = media_reference
+
+
+    # 6. Append the Clip to the Track
+    track.append(clip)
+
+
+
+    # 7. Process the youtube description and insert Markers into the timeline
+    description_file = "tmp/" + youtubeURL + ".description"
+    chapters = process_youtube_description(description_file) 
+
+    markers = create_markers(chapters, dictMeta['fps'])
+
+
+    for marker in markers:
+        clip.markers.append(marker)
+
+
+
+
+
+    #save the timeline as .otio file 
+
+    otio_filename = youtubeURL + ".otio"
+    otio.adapters.write_to_file(timeline, otio_filename)
+    print(
+        "SAVED: {0} with {1} clips.".format(
```

Comment:
since you're generating a track with markers, might make more sense to print the number of markers rather than the number of clips (Which should always be 1?)